libr/util/sys.c

```diff
@@ -218,6 +218,27 @@ R_API ut64 r_sys_now(void) {
 	return ret;
 }
 
+R_API ut64 r_sys_now_mono(void) {
+#if __WINDOWS__
+	LARGE_INTEGER f;
+	if (!QueryPerformanceFrequency (&f)) {
+		return 0;
+	}
+	LARGE_INTEGER v;
+	if (!QueryPerformanceCounter (&v)) {
+		return 0;
+	}
+	v.QuadPart *= 1000000;
+	v.QuadPart /= f.QuadPart;
+	return v.QuadPart;
+#else
+	struct timespec now;
+	clock_gettime (CLOCK_MONOTONIC, &now);
```

Comment:
I still see issues here:

* on modern unix, timespec/timeval is 64bit for seconds, this means we cant fit seconds+usecs in 64bit numbers
* there's an overflow in this multiplication
* we should use the timeradd/timersub/timercmp posix macros
* the CLOCK_MONOTONIC is not everywhere, it must be checked at configure stage or with macros carefully chking each OS version

```
#if __APPLE__ && defined(MAC_OS_X_VERSION_10_12)
```

mac manpage says that:

```
HISTORY
     These functions first appeared in Mac OSX 10.12

STANDARDS
     The clock_gettime(), clock_settime(), and clock_getres() system calls conform to IEEE Std
     1003.1b-1993 (``POSIX.1'').  cleck_gettime_nsec_np() is a non-portable Darwin extension.  The
     clock IDs CLOCK_MONOTONIC_RAW and CLOCK_UPTIME_RAW are extensions to the POSIX interface.
```

this post shows an alternative for pre-monotonic systems, but it depends on globals, that imho can be defined as static to avoid scope polution.

* https://stackoverflow.com/questions/5167269/clock-gettime-alternative-in-mac-os-x

Here's a better alternative:

* https://gist.github.com/jbenet/1087739

Reply:
Note also:

https://github.com/pineapplemachine/PosixMachTiming/tree/improve-degenerate-case-behavior